source/mir/random/discrete.d

```diff
@@ -67,8 +67,9 @@ struct Discrete(T)
     */
     this(const(T)[] cdPoints)
     {
+        import std.range : assumeSorted;
         _cdPoints = cdPoints;
-        r = typeof(r)(_cdPoints);
+        _cdSortedRange = _cdPoints.assumeSorted!"a <= b";
```

Comment:
I still feel it's an unnecessary complication of the implementation to have both `_cdPoints` and `_cdSortedRange`.  The latter is all that's actually _needed_, is there anything else that isn't just workaround for the lack of `const` support?